modules/select2_publish/js/select2.publish.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @file
+ * Select2 integration.
+ */
+(function ($, drupalSettings) {
+  'use strict';
+
+  Drupal.behaviors.select2_publish = {
+    attach: function (context) {
+      $('.select2-widget', context).on('select2-init', function (e) {
+        var config = $(e.target).data('select2-config');
+        config.createTag = function (params) {
+          var term = $.trim(params.term);
+          if (term === '') {
+            return null;
+          }
+          return {
+            id: '$ID:' + term,
+            text: term,
+            published: $(e.target).data('select2-publish-default')
+          };
+        };
+        config.templateSelection = config.templateResult = function (option, item) {
+          if (item) {
+            var published = (option.published === true || $(option.element).attr('data-published') === 'true');
```

Comment:
Entities without publish status (not instance of EntityPublishedInterface) should be styled as published entities.

Reply:
The JS is not attached, if the entity type doesn't implement EntityPublishedInterface